lib/use_case/lodge_assessment.rb

```diff
@@ -177,5 +189,22 @@ def default_address_id(assessment)
       end
       default_address_id
     end
+
+    def address_base_has_uprn?(uprn)
+      !@address_base_search_gateway.search_by_uprn(uprn).empty?
+    end
+
+    def find_assessment_by_id(assessment_id)
+      @assessments_search_gateway.search_by_assessment_id(assessment_id, false)
+        .first
+    end
+
+    def get_assessments_address_id_source(current_address_id:, new_address_id:)
+      if current_address_id == new_address_id
+        "lodgement"
+      else
+        "adjusted_at_lodgement"
```

Comment:
I think we could use something more specific, what about related_address_id or resolved_rrn?